### PR TITLE
refactor: use foundry template

### PR DESCRIPTION
**Pull request by 0x-r4bbit**
The basis is there, but I still need to look into deployment scripts for this one.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[submodule "lib/forge-std"]
+  branch = "v1"
+	path = lib/forge-std
+	url = https://github.com/foundry-rs/forge-std
+[submodule "lib/openzeppelin-contracts"]
+	path = lib/openzeppelin-contracts
+	url = https://github.com/OpenZeppelin/openzeppelin-contracts
+[submodule "lib/minime"]
+	path = lib/minime
+	url = https://github.com/vacp2p/minime
```
Comment (0x-r4bbit) [inline .gitmodules:10]: This uses minime @ https://github.com/vacp2p/minime/commit/0a23a2440dc2f88f59d064c7138363d2e51f5a9a

File: contracts/Semver.sol [hunk 2]
```diff
@@ -31,15 +27,14 @@ contract Semver {
...
+                ".",
+                Strings.toString(MINOR_VERSION),
+                ".",
+                Strings.toString(PATCH_VERSION)
+            )
+        );
     }
-}
\ No newline at end of file
+}
```
Comment (0x-r4bbit) [inline contracts/Semver.sol:40]: Just formatting changes in here

File: package.json [hunk 3]
```diff
@@ -1,16 +1,30 @@
 {
-  "name": "optimism-bridge-snt",
+  "name": "@vacp2p/foundry-template",
+  "description": "Foundry-based template for developing Solidity smart contracts used by Vac",
```
Comment (0x-r4bbit) [inline package.json:3]: Need to update this

File: slither.config.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "detectors_to_exclude": "naming-convention,reentrancy-events,solc-version,timestamp",
+  "filter_paths": "(lib|test)",
+  "solc_remaps": [
+    "@openzeppelin/contracts=lib/openzeppelin-contracts/contracts/",
+    "forge-std/=lib/forge-std/src/"
```
Comment (0x-r4bbit) [inline slither.config.json:6]: This misses the remaps for `minime`

**Review by 3esmit — CHANGES_REQUESTED** [review]
keep other files, just remove whats inside token folder.

**Pull request merged**